### DISCUSSION TITLE
node_module/ng-zorro-antd/tree-view/toggle.d.ts error

### DISCUSSION THREAD

**Discussion by ksrae** [NG-ZORRO/ng-zorro-antd#8362] [Q&A]
Environment: 
Angular 17.1.0
Agnular/cdk 17.1.0
ng-zorro-antd: 17.1.0

I have upgrade my angular project from 13 to 17 and updating ng-zorro 13.4 to 17 as well.
Then I got an error that:
-
node_module/ng-zorro-antd/tree-view/toggle.d.ts - error: TS2611: 'recursive' is defined as a property in class 'CdkTreeNodeToggle<T,T>', but is overridden here in 'NzTreeNodeToggleDirective<T>' as an accessor.

set recursive(value: boolean);
-

I could see the function in "toggle.d.ts" file.

-
get recursive();
set recursive(value: boolean);
-

How can I get rid of this issue?

Thanks,

**Comment by ParsaArvanehPA** [comment]
Greetings @ksrae,
I am investigating this issue, but I cannot reproduce it. No other reports have been made, so it is likely that the problem is on your end.

Please delete package-lock.json and your node_module folder, and then run npm install.

Also, verify your node version. Angular v17 requires node.js versions: v18.13.0 or higher.

Did you follow the angular update guide when upgrading from v13 to v17? You should have upgraded to v14, then v15, then v16, and then v17. If not, please refer to this [link](https://update.angular.io/).

**Reply by ParsaArvanehPA** [reply]
If the issue persists after following the previous steps, kindly share your package.json file with us.

**Comment by ksrae** [comment]
Someone remains this issue on issue tab. 
[https://github.com/NG-ZORRO/ng-zorro-antd/issues/8364](https://github.com/NG-ZORRO/ng-zorro-antd/issues/8364)

If you see CdkTreeNodeToggle and NzTreeNodeToggleDirective classes, you will figure out what it happens.
